ui/timeline/src/commonMain/kotlin/social/androiddev/timeline/FeedCard.kt

```diff
@@ -0,0 +1,228 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Card
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.material.contentColorFor
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import social.androiddev.common.composables.UserAvatar
+import social.androiddev.common.theme.MastodonTheme
+
+/**
+ * Stateful
+ */
+@Composable
+fun FeedCard(
+    state: FeedItemState,
+    modifier: Modifier = Modifier,
+    contentPadding: PaddingValues = PaddingValues(8.dp),
+    backgroundColor: Color = MaterialTheme.colors.surface,
+    contentColor: Color = contentColorFor(backgroundColor),
+) {
+    FeedCard(
+        userAvatarUrl = state.userAvatarUrl,
+        date = state.date,
+        username = state.username,
+        userAddress = state.acctAddress,
+        toot = state.message,
+        videoUrl = state.videoUrl,
+        images = state.images,
+        contentPadding = contentPadding,
+        modifier = modifier,
+        backgroundColor = backgroundColor,
+        contentColor = contentColor
+    )
+}
+
+/**
+ * Stateless
+ */
+@Composable
+fun FeedCard(
+    userAvatarUrl: String,
+    date: String,
+    username: String,
+    userAddress: String,
+    toot: String?,
+    videoUrl: String?,
+    images: List<String>,
+    modifier: Modifier = Modifier,
+    contentPadding: PaddingValues = PaddingValues(8.dp),
+    backgroundColor: Color = MaterialTheme.colors.surface,
+    contentColor: Color = contentColorFor(backgroundColor),
+) {
+    Card(
+        modifier = modifier,
+        backgroundColor = backgroundColor,
+        contentColor = contentColor,
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .wrapContentHeight()
+                .padding(contentPadding),
+            verticalAlignment = Alignment.CenterVertically,
+        ) {
+            UserAvatar(
+                modifier = Modifier
+                    .size(72.dp)
+                    .clip(CircleShape),
+                url = userAvatarUrl
+            )
+
+            HorizontalSpacer()
+
+            TootContent(
+                modifier = Modifier.weight(1f).wrapContentHeight(),
+                username = username,
+                userAddress = userAddress,
+                message = toot,
+                date = date,
+                images = images,
+                videoUrl = videoUrl
+            )
+        }
+    }
+}
+
+@Composable
+private fun TootContent(
```

Comment:
I would extract this in a separate file to reduce the size here

Reply:
Agreed - will move this and the TopLabel to a different file and rename to something more general.